api/envoy/config/filter/network/http_connection_manager/v2/http_connection_manager.proto

```diff
@@ -171,6 +171,21 @@ message HttpConnectionManager {
   // :ref:`config_http_conn_man_headers_x-forwarded-for` for more information.
   uint32 xff_num_trusted_hops = 19;
 
+  // If set, Envoy will not append the remote address to the
+  // :ref:`config_http_conn_man_headers_x-forwarded-for` HTTP header. This may be used in
+  // conjunction with HTTP filters that explicitly manipulate XFF after the HTTP connection manager
+  // has mutated the request headers.
+  bool skip_xff_append = 21;
+
+  // Do not consider requests to ever be of internal provenance.
```

Comment:
nit: Can we link into the docs into relevant places where we have discussion of sanitizing, XFF, etc.?

Reply:
This is a quick workaround for Envoy's access permissions not working for us and is made possible because it doesn't do anything we need (today).    There's comments around the existing internal_request logic that it's probably going to have to be pluggable / configurable already so I just want to make sure we're sufficiently future thinking here.  

If we think there's value in our internal "Is this trusted" sharing code with Envoy's existing "is this trusted" we may be better off making the "is this trusted" logic pluggable, checking in 2 reference models (backwards compatbile mode, and always return false mode) and building with the latter (and then future uses can build with complicated-internal-existing-logic).    If we think that Envoy's "is this trusted" is never needed for our and (some) other's downstream use, we can keep this as-is as it's way simpler and then we don't have to constantly monitor features under the "trusted" umbrella to make sure we're truly Ok with them.

Reply:
Do we want to distinguish between "trusted" and "internal" here? I think we probably do want to make these concept(s) pluggable eventually, but agree that it is more expedient and probably generally useful to be able to set this globally as well. @mattklein123 do you think it's worth adding extension points for determining this given we don't currently have a strong motivation on our side?

Reply:
I don't think we should add extension points unless there is a real need for them. IMO the existing Envoy logic is probably good enough for almost everyone, especially if we add some global knobs, so I'm fine with this as long as @alyssawilk is also OK with that for now.